From 833e3308139d50c819114c0927a9ecf9f62584e9 Mon Sep 17 00:00:00 2001
From: "GL.iNet-XiBao.Lu" <xibao.lu@gl-inet.com>
Date: Wed, 21 Jun 2023 01:19:01 -0700
Subject: [PATCH] dts: fix dts for gl-ax1800

merge patchs:
0005-ipq807x-fit-dts-for-gl-ax-t-1800.patch
0018-fix-ax1800-for-i2c-gpio-custom.patch
0021-fix-axt1800-add-radio-config-to-gl-hw-info.patch
0022-ax1800-dts-add-ble-uart.patch
0028-add-usb-reset-to-dts.patch

Signed-off-by: GL.iNet-XiBao.Lu <xibao.lu@gl-inet.com>
---
 .../boot/dts/qcom/qcom-ipq6018-gl-ax1800.dts  | 42 ++++++++++++++
 .../boot/dts/qcom/qcom-ipq6018-gl-ax1800.dtsi | 57 +++++++++++++++----
 2 files changed, 87 insertions(+), 12 deletions(-)

diff --git a/feeds/ipq807x/ipq807x/files/arch/arm64/boot/dts/qcom/qcom-ipq6018-gl-ax1800.dts b/feeds/ipq807x/ipq807x/files/arch/arm64/boot/dts/qcom/qcom-ipq6018-gl-ax1800.dts
index 77682246..ad328120 100644
--- a/feeds/ipq807x/ipq807x/files/arch/arm64/boot/dts/qcom/qcom-ipq6018-gl-ax1800.dts
+++ b/feeds/ipq807x/ipq807x/files/arch/arm64/boot/dts/qcom/qcom-ipq6018-gl-ax1800.dts
@@ -22,11 +22,19 @@
 	compatible = "glinet,ax1800", "qcom,ipq6018-cp03", "qcom,ipq6018";
 
 	aliases {
+		serial0 = &blsp1_uart3; //system uart = ttyMSM0
+		serial1 = &blsp1_uart4; //tl8251      = ttyMSM1
 		ethernet3 = "/soc/dp4";
 		ethernet4 = "/soc/dp5";
 	};
 };
 
+&gl_hw {
+	model = "ax1800";
+	lan = "eth1 eth2 eth3 eth4";
+	usb-port = "1-1,2-1";
+};
+
 &mdio0 {
 	phy3: ethernet-phy@3 {
 		reg = <3>;
@@ -52,6 +60,16 @@
 };
 
 &soc {
+	blsp1_uart4: serial@78b2000 {
+		compatible = "qcom,msm-uartdm-v1.4", "qcom,msm-uartdm";
+		reg = <0x78b2000 0x200>;
+		interrupts = <GIC_SPI 307 IRQ_TYPE_LEVEL_HIGH>;
+		clocks = <&gcc GCC_BLSP1_UART4_APPS_CLK>,
+			<&gcc GCC_BLSP1_AHB_CLK>;
+		clock-names = "core", "iface";
+		status = "disabled";
+	};
+
 	dp4 {
 		device_type = "network";
 		compatible = "qcom,nss-dp";
@@ -76,3 +94,27 @@
 		phy-mode = "sgmii";
 	};
 };
+
+&tlmm {
+	uart_pins: uart_pins {
+		mux_1 {
+			pins = "gpio42", "gpio43"; // i2c
+			function = "gpio";
+			bias-pull-up;
+		};
+	};
+	uart_tl8251_pins: uart_tl8251_pins {
+		mux {
+			pins = "gpio75", "gpio76"; // gpio 77 reset tl8251
+			function = "blsp3_uart";
+			drive-strength = <8>;
+			bias-pull-down;
+		};
+	};
+};
+
+&blsp1_uart4 {
+	pinctrl-0 = <&uart_tl8251_pins>;
+	pinctrl-names = "default";
+	status = "ok";
+};
diff --git a/feeds/ipq807x/ipq807x/files/arch/arm64/boot/dts/qcom/qcom-ipq6018-gl-ax1800.dtsi b/feeds/ipq807x/ipq807x/files/arch/arm64/boot/dts/qcom/qcom-ipq6018-gl-ax1800.dtsi
index 25f8717a..6b6e34bb 100644
--- a/feeds/ipq807x/ipq807x/files/arch/arm64/boot/dts/qcom/qcom-ipq6018-gl-ax1800.dtsi
+++ b/feeds/ipq807x/ipq807x/files/arch/arm64/boot/dts/qcom/qcom-ipq6018-gl-ax1800.dtsi
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019, The Linux Foundation. All rights reserved.
+ * Copyright (c) 2018-2020, The Linux Foundation. All rights reserved.
  *
  * Permission to use, copy, modify, and/or distribute this software for any
  * purpose with or without fee is hereby granted, provided that the above
@@ -28,16 +28,37 @@
 		ethernet1 = "/soc/dp2";
 		ethernet2 = "/soc/dp3";
 
-		led-boot = &led_run;
-		led-failsafe = &led_run;
-		led-running = &led_run;
-		led-upgrade = &led_run;
+		//led-boot = &led_run;
+		//led-failsafe = &led_run;
+		//led-running = &led_run;
+		//led-upgrade = &led_run;
 	};
 
 	chosen {
 		bootargs = "console=ttyMSM0,115200,n8 rw init=/init";
 		bootargs-append = " swiotlb=1 coherent_pool=2M";
 	};
+
+	gl_hw: gl_hw {
+		compatible = "gl-hw-info";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		model = "axt1800";
+		wan = "eth0";
+		reset-button = "gpio-50";
+		usb-port = "1-1,2-1";
+		usb-reset = "usb_power,1";
+		flash_size = <128>;
+		temperature = "/sys/devices/virtual/thermal/thermal_zone0/temp";
+		radio = "radio0";
+		factory_data {
+			device_mac = "0:ART", "0x0";
+			device_ddns = "0:ART", "0x20";
+			device_sn_bak = "0:ART", "0x30";
+			device_sn = "0:ART", "0x40";
+			country_code = "0:ART", "0x98";
+		};
+	};
 };
 
 &tlmm {
@@ -101,7 +122,7 @@
 		};
 		qpic_pad {
 			pins = "gpio1", "gpio3", "gpio4",
-			       "gpio10", "gpio11", "gpio17";
+				   "gpio10", "gpio11", "gpio17";
 			function = "qpic_pad";
 			drive-strength = <8>;
 			bias-pull-down;
@@ -250,14 +271,14 @@
 		pinctrl-0 = <&leds_pins>;
 		pinctrl-names = "default";
 
-		led_run: led@35 {
-			label = "white:sys";
+		led_run: white {
+			label = "white_led";
 			gpios = <&tlmm 35 GPIO_ACTIVE_HIGH>;
-			default-state = "on";
+			default-state = "off";
 		};
 
-		led@37 {
-			label = "blue:wan";
+		blue {
+			label = "blue_led";
 			gpios = <&tlmm 37 GPIO_ACTIVE_HIGH>;
 			default-state = "on";
 		};
@@ -271,7 +292,7 @@
 		switch {
 			label = "switch";
 			linux,code = <KEY_WPS_BUTTON>;
-			gpios = <&tlmm 9 GPIO_ACTIVE_LOW>;
+			gpios = <&tlmm 9 GPIO_ACTIVE_HIGH>;
 			linux,input-type = <1>;
 			debounce-interval = <60>;
 		};
@@ -283,6 +304,18 @@
 			debounce-interval = <60>;
 		};
 	};
+
+	gpio-export {
+		compatible = "gpio-export";
+		pinctrl-0 = <&usb_pins>;
+		pinctrl-names = "default";
+
+		usb_power {
+			gpio-export,name = "usb_power";
+			gpio-export,output = <1>;
+			gpios = <&tlmm 0 GPIO_ACTIVE_HIGH>;
+		};
+	};
 };
 
 &blsp1_uart3 {
-- 
2.25.1

